sdk/subscribe_block.go

```diff
@@ -24,6 +25,96 @@ const (
 	pathCosignature        = "cosignature"
 )
 
+// Closes the subscription channel.
+func closeChannel(s *subscribe) {
+	switch s.Ch.(type) {
+	case chan *BlockInfo:
+		chType := s.Ch.(chan *BlockInfo)
+		close(chType)
+
+	case chan *StatusInfo:
+		chType := s.Ch.(chan *StatusInfo)
+		delete(statusInfoChannels, s.getAdd())
+		close(chType)
+
+	case chan *HashInfo:
+		chType := s.Ch.(chan *HashInfo)
+		delete(unconfirmedRemovedChannels, s.getAdd())
+		close(chType)
+
+	case chan *PartialRemovedInfo:
+		chType := s.Ch.(chan *PartialRemovedInfo)
+		delete(partialRemovedInfoChannels, s.getAdd())
+		close(chType)
+
+	case chan *SignerInfo:
+		chType := s.Ch.(chan *SignerInfo)
+		delete(signerInfoChannels, s.getAdd())
+		close(chType)
+
+	case chan *ErrorInfo:
+		chType := s.Ch.(chan *ErrorInfo)
+		delete(errChannels, s.getAdd())
+		close(chType)
+
+	default:
+		chType := s.Ch.(chan Transaction)
```

Comment:
Type 'chan Transaction' checking should be done. We can get a panic when changing the code here.
default-case need return error 'WRONG TYPE CHANNEL'.
Also we need add case 'nil'.